### DISCUSSION TITLE
inherit_from not working for custom cop gem

### DISCUSSION THREAD

**Discussion by renjiexu** [rubocop/rubocop#9717] [Q&A]
```
# default.yml

inherit_from:
  - rubocop_custom_cop.yml

AllCops:
  ...
```

```
# rubocop_custom_cop.yml

Company/MagicCop
  Enabled: true
  Params: ['abc']
```

Hi team, I've been using https://github.com/rubocop/rubocop-extension-generator to generate custom cop gem, besides everything generated by default, I separated custom cops out from `default.yml` to `rubocop_custom_cop.yml`, however, seems it's not been able to load anything from the `rubocop_custom_cop.yml`.

By reading the config related code, it seems the only thing I need to do is: https://github.com/renjiexu/rubocop-extension-generator/commit/7b5741d2141dbfac72787774ae7ac7e46bc92724.

This seems to be a bug of rubocop-extension-generator but the `inject.rb` was widely used in various `rubocop-extensions` gems, would like to confirm this before submitting the PR.

Thanks 🙏

**Comment by koic** [comment]
The following patch cause the following errors:

```diff
diff --git a/lib/rubocop/performance/inject.rb b/lib/rubocop/performance/inject.rb
index e255a881a..9bdde53d0 100644
--- a/lib/rubocop/performance/inject.rb
+++ b/lib/rubocop/performance/inject.rb
@@ -7,7 +7,7 @@ module RuboCop
     module Inject
       def self.defaults!
         path = CONFIG_DEFAULT.to_s
-        hash = ConfigLoader.send(:load_yaml_configuration, path)
+        hash = ConfigLoader.send(:load_file, File.new(path))
         config = Config.new(hash, path).tap(&:make_excludes_absolute)
         puts "configuration from #{path}" if ConfigLoader.debug?
         config = ConfigLoader.merge_with_default(config, path)
```

I think this is an unexpected behavior.

```console
% bundle exec rubocop
Error: unrecognized cop Performance/AncestorsInclude found in performance/config/default.yml
unrecognized cop Performance/ArraySemiInfiniteRangeSlice found in performance/config/default.yml
unrecognized cop Performance/BigDecimalWithNumericArgument found in performance/config/default.yml
unrecognized cop Performance/BindCall found in performance/config/default.yml
unrecognized cop Performance/BlockGivenWithExplicitBlock found in performance/config/default.yml
(snip)
```

So, I think rubocop-extension-generator is designed to write custom cops configuration in config/default.yml of custom gem.
The following configuration should be placed in the config/default.yml of your custom gem.

```yaml
# config/default.yml
Company/MagicCop
  Enabled: true
  Params: ['abc']
```

e.g. config/default.yml of RuboCop Performance:
https://github.com/rubocop/rubocop-performance/blob/v1.11.0/config/default.yml
